atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTableTest.java

```diff
@@ -347,6 +348,104 @@ public void createDoesNotThrowWhenOverflowColumnAlreadyExistsAndAlterIsRan() thr
         verifyTableAltered();
     }
 
+    @Test
+    public void createThrowsIfAlterConfiguredAndMetadataNeedsUpdateAndMigrationInProgress() {
+        assertThatThrownBy(() -> testAlterMetadataForMigrationState(OverflowMigrationState.IN_PROGRESS))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageContaining(MISSING_OVERFLOW_EXCEPTION_MESSAGE);
+        verifyTableSizeMetadataNotUpdated();
+    }
+
+    @Test
+    public void createThrowsIfAlterConfiguredAndMetadataNeedsUpdateAndMigrationUnstarted() {
+        assertThatThrownBy(() -> testAlterMetadataForMigrationState(OverflowMigrationState.UNSTARTED))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageContaining(MISSING_OVERFLOW_EXCEPTION_MESSAGE);
+        verifyTableSizeMetadataNotUpdated();
+    }
+
+    @Test
+    public void createThrowsIfAlterConfiguredAndMetadataNeedsUpdateAndMigrationFinishing() {
+        assertThatThrownBy(() -> testAlterMetadataForMigrationState(OverflowMigrationState.FINISHING))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageContaining(MISSING_OVERFLOW_EXCEPTION_MESSAGE);
+        verifyTableSizeMetadataNotUpdated();
+    }
+
+    @Test
+    public void createAltersMetadataIfConfiguredAndOverflowTableExists() throws TableMappingNotFoundException {
+        testAlterMetadataForMigrationState(OverflowMigrationState.FINISHED);
+        verifyTableSizeMetadataUpdated();
+    }
+
+    private void testAlterMetadataForMigrationState(OverflowMigrationState state) throws TableMappingNotFoundException {
```

Comment:
a bit uncertain of the naming here: test?

Reply:
fair fair, previously it had asserts/verify, but I moved them later on.

Alternatively,  I think we can get rid of these need for these tests if we are ok with just testing the helper method!